File: ruby/client_async.rb
require 'celluloid/io'
require 'http'

class NapFetcher
  include Celluloid::IO

  def request_nap(msg)
    HTTP.get(
      "http://localhost:4567?sec=3&msg=#{msg}", 
      socket_class: Celluloid::IO::TCPSocket
    )
  end
  
end

fetcher = NapFetcher.new

words = %w(i am fast)

futures = []

words.each do |word|
  futures << fetcher.future.request_nap(word)
end

futures.each do |future|
  response = future.value
  puts response.body
end

File: python/client_async.py
import tornado.ioloop
import tornado.httpclient

ioloop = tornado.ioloop.IOLoop.instance()

def handle_request(response):
    if response.error:
        print("Error:", response.error)
    else:
        print(response.body)

http_client = tornado.httpclient.AsyncHTTPClient()

words = ['i', 'am', 'fast']

for word in words:
  http_client.fetch("http://localhost:4567?sec=3&msg=%s" % (word), handle_request)

ioloop.start()
